cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -312,17 +322,110 @@ public void testExternalSparkProgramPipelines() throws Exception {
     testMacroEvaluationActionPipeline(Engine.MAPREDUCE);
+    // Deploy two pipelines which are scheduled to be triggered by the completion of the pipeline deployed in
+    // testMacroEvaluationActionPipeline(Engine.SPARK)
+    WorkflowManager triggeredWorkflowManagerMR = deployPipelineWithSchedule(Engine.MAPREDUCE);
+    WorkflowManager triggeredWorkflowManagerSpark = deployPipelineWithSchedule(Engine.SPARK);
     testMacroEvaluationActionPipeline(Engine.SPARK);
+    // After the completion of the above pipeline, verify the results of two triggered pipelines
+    assertTriggeredPipelinesResult(triggeredWorkflowManagerMR, Engine.MAPREDUCE);
+    assertTriggeredPipelinesResult(triggeredWorkflowManagerSpark, Engine.SPARK);
+  }
+
+  private WorkflowManager deployPipelineWithSchedule(Engine engine) throws Exception {
+    String tableName = "actionScheduleTable" + engine;
+    String key1 = "trigger-runtime-arg";
+    String key2 = "trigger-plugin-property";
+    String sourceName = "macroActionWithScheduleInput-" + engine;
+    String sinkName = "macroActionWithScheduleOutput-" + engine;
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      // 'filter' stage is configured to remove samuel, but action will set an argument that will make it filter dwayne
+      .addStage(new ETLStage("action1", MockAction.getPlugin(tableName, "row1", "column1",
+                                                             String.format("${%s}", key1))))
+      .addStage(new ETLStage("action2", MockAction.getPlugin(tableName, "row2", "column2",
+                                                             String.format("${%s}", key2))))
+      .addStage(new ETLStage("source", MockSource.getPlugin(sourceName)))
+      .addStage(new ETLStage("filter1", StringValueFilterTransform.getPlugin("name", String.format("${%s}", key1))))
+      .addStage(new ETLStage("filter2", StringValueFilterTransform.getPlugin("name", String.format("${%s}", key2))))
+      .addStage(new ETLStage("sink", MockSink.getPlugin(sinkName)))
+      .addConnection("action1", "action2")
+      .addConnection("action2", "source")
+      .addConnection("source", "filter1")
+      .addConnection("filter1", "filter2")
+      .addConnection("filter2", "sink")
+      .setEngine(engine)
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("triggeredActionPipeline-" + engine);
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    // there should be only two programs - one workflow and one mapreduce/spark
+    Schema schema = Schema.recordOf("testRecord", Schema.Field.of("name", Schema.of(Schema.Type.STRING)));
+    StructuredRecord recordSamuel = StructuredRecord.builder(schema).set("name", "samuel").build();
+    StructuredRecord recordKey1Value = StructuredRecord.builder(schema).set("name", "macroValue").build();
```

Comment:
this is not good, you have to read another method in order to understand why you have 'macroValue' here, and now your two methods are invisibly coupled.